src/integrations/stripe.ts

```diff
@@ -368,10 +368,18 @@ export async function checkForPaymentCheckoutCompletion(
     featheryWindow().history.replaceState({}, '', newUrl);
 
     // search all step buttons and subgrids for the purchase action and complete the payment
-    const paymentElement = [...step.buttons, ...step.subgrids].find((element) =>
-      element.properties.actions?.find(
-        (action: any) => action.type === ACTION_PURCHASE_PRODUCTS
-      )
+    const paymentElement: any = Object.values(steps).reduce(
```

Comment:
why does this need to be a reduce? versus .find

Reply:
Find on steps would not work since what this bit of code does is find particular element in one of the steps.  Reduce seems appropriate to me.